### PR TITLE
refactor code

### PR DESCRIPTION
**Pull request by peterboro**
# Dear Reviewer(s)

## Thank you for taking the time to review this pull request.

I implemented the following:
- Make sure YAGNI, KISS, and DRY rules and Ruby style guide are respected.
- Follow SOLID principles.
- Pay special attention to the "Single-responsibility Principle".
    - Make sure that the role of `main.rb` is only to start your code and ask the user for the initial choice of options (or to quit).
    - Extract some pieces of logic into a separate class(es).

### CODE REVIEW

File: options.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require_relative 'app'
+
```
Comment (HAMMAS-SALEEM) [inline options.rb:2]: - [ ] As you guys know, this is a pair-programming project but i cannot see any commit from the second member of the team 🤔 Kindly ask your partner to make a commit in this pull request 😊

**Review by HAMMAS-SALEEM — COMMENTED** [review]
# Invalid Review Request 🔴

## Hey @peterboro :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following some requirements

Good job on implementing some of the requirements, I cannot proceed review on your pull request there are some issues that you need to work on :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 2]
```diff
@@ -2,8 +2,9 @@
 require_relative 'teacher'
 require_relative 'student'
 require_relative 'rental'
+require_relative 'user_inputs'
```
Comment (HAMMAS-SALEEM) [inline app.rb:6]: image

- [ ] Your methods should be SOLID means single responsible but you can see that your methods are checking the input(First Responsibility), taking the inputs(Second Responsibility), creating the like in this case creating a person class(Third Responsibility) from taken inputs. :thinking: Kindly make all your methods SOLID :+1:

Comment (HAMMAS-SALEEM) [inline app.rb:6]: image

- [ ] You are writing **print** and **gets.chomp** a lot of times means you are repeating your code which is not making your code DRY. Kindly make your code DRY by making a function for taking inputs from the users :+1:

Hint: See the function below ⬇️
```
def inputs(show)
  outputs = []
  show.each do |s|
    print s
    outputs.push(gets.chomp)
  end
  outputs
end
```

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

## Hey @peterboro, @shyusu4 :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following some requirements

Good job on implementing some of the requirements, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by peterboro** [comment]
Dear Reviewer. 

Kindly review my pull request. 
MarcoStudentSuccess

**Review by bellom — APPROVED** [review]
@peterboro,

### This is a manual review

Good job refactoring code 👍 

Note: It will be approved on your dashboard as soon as possible.

# Approved :+1:

Cheers and Happy coding!:clap::clap::clap: (edited)

**Comment by peterboro** [comment]
Thank you so much.

**Pull request merged**